### DISCUSSION TITLE
Issues using pyinstaller and spacy 3.0.6 (en_core_web_ssm.load())

### DISCUSSION THREAD

**Discussion by JimMcDonough** [explosion/spaCy#8311] [Help: Coding & Implementations] [answered]
I have had issues with this en_core_web_sm.load() in the past.  The help I found online was helpful but this new version has different issues.  The error I am getting is below.  I am thinking that the spacy_legacy architecture is not be found and resolved.  I am looking for a better understanding how spacy uses the legacy_legacy architectures and where spacy stores its architectures.

catalogue.RegistryError: [E893] Could not find function 'spacy.Tok2Vec.v1' in function registry 'architectures'. If you're using a custom function, make sure the code is available. If the function is provided by a third-party package, e.g. spacy-transformers, make sure the package is installed in your environment.

Available names: spacy.CharacterEmbed.v2, spacy.EntityLinker.v1, spacy.HashEmbedCNN.v2, spacy.MaxoutWindowEncoder.v2, spacy.MishWindowEncoder.v2, spacy.MultiHashEmbed.v2, spacy.PretrainCharacters.v1, spacy.PretrainVectors.v1, spacy.Tagger.v1, spacy.TextCatBOW.v1, spacy.TextCatCNN.v1, spacy.TextCatEnsemble.v2, spacy.TextCatLowData.v1, spacy.Tok2Vec.v2, spacy.Tok2VecListener.v1, spacy.TorchBiLSTMEncoder.v1, spacy.TransitionBasedParser.v1, spacy.TransitionBasedParser.v2
[9632] Failed to execute script DDP_GUI_JM_Drop_Down_Prototype_11_16_20

**Comment by svlandeg** [comment]
[`spacy-legacy`](https://github.com/explosion/spacy-legacy) is a separate package, and should be installed when you install `spacy` v3. Do you have it installed in your environment?

**Comment by polm** [comment]
> I am looking for a better understanding how spacy uses the legacy_legacy architectures and where spacy stores its architectures.

Architectures are stored in the "registry", which you can think of as an internal dictionary in spaCy. The main way things are added to the registry is by using a decorator on functions. The decorator is executed when the function is defined, so if it's in spacy-legacy it would be added when that package was imported, for example. 

Check the [registry docs](https://spacy.io/api/top-level#registry) for more info.

**Reply by JimMcDonough** [reply]
Thank you

**Comment by JimMcDonough** [comment, marked as answer]
Solved my issue
https://github.com/pyinstaller/pyinstaller/discussions/5896#discussioncomment-842074
